### PR TITLE
GO1-11453: Course Assigned Through A Group Email Notification Not Received

### PR DESCRIPTION
**Pull request by nhanai87**

### CODE REVIEW

File: plan/PlanRepository.php [hunk 1]
```diff
@@ -164,8 +164,8 @@ public function create(Plan &$plan, bool $notify = false, array $queueContext =
         ]);
 
         $plan->id = $this->db->lastInsertId('gc_plan');
-        $plan->notify = $notify;
-        $queueContext['notify'] = $notify;
+        $plan->notify = $notify ?: $queueContext['notify'];
```
Comment (drushman) [inline plan/PlanRepository.php]: What if $queueContext does not have 'notify' key?

Comment (nhanai87) [reply plan/PlanRepository.php]: yep, completely make sense,
What do you think if I change to "$plan->notify = $queueContext['notify'] ?? $notify;" ?

Comment (pvmchau) [reply plan/PlanRepository.php]: "$notify" has higher priority than context value

Comment (pvmchau) [reply plan/PlanRepository.php]: @drushman we can add isset() checking for "$queueContext['notify']" i think

**Comment by binhminhgo1** [comment]
@drushman duy and @pvmchau It's look good for me.

File: plan/PlanRepository.php [hunk 2]
```diff
@@ -164,8 +164,8 @@ public function create(Plan &$plan, bool $notify = false, array $queueContext =
         ]);
 
         $plan->id = $this->db->lastInsertId('gc_plan');
-        $plan->notify = $notify;
-        $queueContext['notify'] = $notify;
+        $plan->notify = $notify ?: $queueContext['notify'] ?? false;
```
Comment (drushman) [inline plan/PlanRepository.php]: Can you please add group statements by `()`?

Comment (nhanai87) [reply plan/PlanRepository.php]: updated @drushman

**Pull request merged**